documentation/high-level-introduction.md

```diff
@@ -0,0 +1,257 @@
+# A brief introduction to msbuild (Level 101)
+
+## Introduction
+
+This document aims to be a “101 college” level introduction into msbuild, useful for those coming to msbuild with very little background in how it works.
+
+### What is msbuild?
+
+msbuild is a build system commonly used to build .NET projects. It is made up the msbuild toolchain itself, and some number of additional files provided by the SDKs which provide platform specific instructions on how to process projects of those types.
+
+msbuild is built up of a few very generic ideas:
+
+- Run a sequence of defined steps based upon input / output files and dependency ordering
+- Store single and groups of values in a global table
```

Comment:
Because they _are_ separate tables. If you define one as a group and try to read it as a single you'll get empty string as a return.

The fact that you could have two different variables with the same name was the source of no end of trouble.

Reply:
Are you talking about properties vs. items? Other than that they're defined and accessed differently, they're essentially the same thing. You could have a property with lots of semicolons, for instance. You're right that it could be confusing, so perhaps it's best to say "Store values in one of _two_ global tables"? That would clarify that the tables are fully separate, so you can have two variables with the same name.